adagrams/game.py

```diff
@@ -1,13 +1,80 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2,
+    'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2,
+    'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1,
+    'Y': 2, 'Z': 1
+}
+
 def draw_letters():
-    pass
+
+    # A temporary pool built from LETTER_POOL counts
+    pool = []
+    for letter, quantity in LETTER_POOL.items():
+        pool += [letter] * quantity
+    
+    hand = []
+    for _ in range(10):
+        idx = randint(0, len(pool) - 1)     # pick a random index
+        hand.append(pool[idx])              # add the letter at that index of pool
+        pool.pop(idx)                       # remove it so it can't be drawn again
+    
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    hand = []
+
+    for item in letter_bank:
+        hand.append(item)
+
+    for char in word.upper():
+        if char in hand:
+            idx = hand.index(char)
+            hand.pop(idx)
+        else:
+            return False
+    
+    return True
 
 def score_word(word):
-    pass
+    score_chart = {
+        'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4,
+        'I': 1, 'J': 8, 'K': 5, 'L': 1, 'M': 3, 'N': 1, 'O': 1, 'P': 3,
+        'Q': 10,'R': 1, 'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8,
+        'Y': 4, 'Z': 10
+    }
```

Comment:
This is a constant variable
```suggestion
    SCORE_CHART = {
        'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4,
        'I': 1, 'J': 8, 'K': 5, 'L': 1, 'M': 3, 'N': 1, 'O': 1, 'P': 3,
        'Q': 10,'R': 1, 'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8,
        'Y': 4, 'Z': 10
    }
```